autospec/infile_handler.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/true
+#
+# infile_parser.py - part of autospec
+# Copyright (C) 2017 Intel Corporation
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import os
+import requests
+import tempfile
+from urllib.request import urlretrieve
+
+import util
+import infile_bb_parser
+
+
+# filetypes and their parse order (lower parsed first)
+parsable_filetypes = {'.inc': 1, '.bb': 2}
+
+
+def parse_ext(path):
+    ext = os.path.splitext(path)[1]
+    if ext not in parsable_filetypes:
+        util.print_warning("Cannot parse infile \"{}\" type. "
+                           "from input: {}".format(ext, path))
+        return
+
+    if ext.startswith('.'):
+        ext = ext[1:]
+
+    return ext
+
+
+def check_url_content(url):
+    if "text/html" in requests.head(url).headers['content-type']:
+        if 'git' in url and "/tree/" in url:
+            url = url.replace("/tree/", "/plain/", 1)
```

Comment:
is this url scheme (tree -> plain) valid for all git hosting services, or just for yoctoproject? It looks like this doesn't work for github.

Reply:
Added github parsing
Done :)